engine/src/execution_graph/logic_controllers/BatchAggregationProcessing.h

```diff
@@ -25,30 +25,58 @@ class ComputeAggregateKernel : public kernel {
 	ComputeAggregateKernel(std::size_t kernel_id, const std::string & queryString, std::shared_ptr<Context> context, std::shared_ptr<ral::cache::graph> query_graph)
 		: kernel{kernel_id, queryString, context, kernel_type::ComputeAggregateKernel} {
         this->query_graph = query_graph;
+
+        std::tie(this->group_column_indices, this->aggregation_input_expressions, this->aggregation_types,
+            this->aggregation_column_assigned_aliases) = ral::operators::parseGroupByExpression(this->expression);
 	}
 
     bool can_you_throttle_my_input() {
 		return true;
 	}
 
 	virtual kstatus run() {
-		CodeTimer timer;
+		
+        CodeTimer timer;
         CodeTimer eventTimer(false);
-
-        std::vector<std::string> aggregation_input_expressions, aggregation_column_assigned_aliases;
-        std::tie(this->group_column_indices, aggregation_input_expressions, this->aggregation_types,
-            aggregation_column_assigned_aliases) = ral::operators::parseGroupByExpression(this->expression);
+    
+        eventTimer.start();
 
         bool ordered = false; // If we start using sort based aggregations this may need to change
-        BatchSequence input(this->input_cache(), this, ordered);
+        BatchSequenceBypass input(this->input_cache(), this); // WSM TODO, need to make BatchSequenceBypass take in "ordered" parameter
         int batch_count = 0;
         while (input.wait_for_next()) {
 
-            this->output_cache()->wait_if_cache_is_saturated();
-            auto batch = input.next();
+            // this->output_cache()->wait_if_cache_is_saturated();
+            std::vector<std::unique_ptr<ral::cache::CacheData>> batches;
```

Comment:
If i wanted to send this batch to multiple destinations what changes would have to be made to this code? I think that it almost cant be a unique_ptr

Reply:
See here https://github.com/BlazingDB/blazingsql/issues/924

Reply:
The short answer is that, correct, you dont want it to be a unique_ptr. This PR is meant to be one small step towards a broader goal. Working forward in smaller steps can help the bigger goal not be one super massive PR